### DISCUSSION TITLE
Why DAX Studio always show Total Size of database different than SQL Server Management Studio?

### DISCUSSION THREAD

**Discussion by Andryno** [DaxStudio/DaxStudio#666] [Q&A]
I compared database size of all models from Properties window in SQL Server Management Studio (v18.8) for my SSAS 2016 instance (v13.0.5882.1) with same info from Summary Page of View Metrics in DAX Studio (v2.16.1). Size values are always different and size in DAX Studio is more than SSMS estimation up to 20%-100% (all databases of 1200 compatibility mode). What is reason of this difference and what tool gives more precise DB size estimation?

**Comment by marcosqlbi** [comment]
DAX Studio uses the VertiPaq Analyzer libraries, which analyze the size of the database structures loaded in memory, obtained through data management views of Analysis Services.
This number is usually bigger than the size reported by SSMS, which is an Estimated Size. Unfortunately, Microsoft does not provide documentation about how that number is estimated.
However, keep in mind that the memory required to run the database is bigger because once loaded in memory the database requires additional memory to store intermediate tables created during table scan - that size depends on the materialization performed by individual queries.

**Reply by Andryno** [reply]
Thanks for answer. Seldom, but in some cases SSMS estimation is bigger than DAX Studio. What is really strange with size of database it's difference between DAX Studio DB size estimation and SSMS backup of the same DB (.abf file). I've processed one DB, made backup of it and see estimations: in DAX Studio View Metrics Summary 678 Mb, in SSMS Properties Window 1605 Mb, abf DB backup made from SSMS 935 Mb. It looks like DAX Studio estimation with some uncrompressed objects in RAM is even less than fully compressed abf backup on disk.

**Reply by marcosqlbi** [reply]
This is weird, the ABF is usually smaller. Because the ABF is a zip, you might want to uncompress it and find where there are 300MB unreported by VertiPaq Analyzer.

**Comment by kpantiga** [comment]
Marco, do you have the DMV query to get the Toal Size of the database in memory (Total Size metric from View Metrics Summary tab)?  

My goal is to get a nice query where I can report my memory usage by database.

Much appreciated

**Reply by marcosqlbi** [reply]
DAX Studio uses VertiPaq Analyzer, which collects data from multiple DMVs and provides an easier way to access the info.
Unfortunately, the process is more complex than just querying a single DMV, and it is the reason why VertiPaq Analyzer requires a separate library.

Your best option is to use the VertiPaq Analyzer library and consume the info like DAX Studio does.
Look at how TotalSize is evaluated in https://github.com/DaxStudio/DaxStudio/blob/master/src/DaxStudio.UI/ViewModels/VertipaqAnalyzer/VpaSummaryViewModel.cs

These are the NuGet libraries:
https://www.nuget.org/packages/Dax.Vpax/
https://www.nuget.org/packages/Dax.ViewModel/
https://www.nuget.org/packages/Dax.Metadata/
https://www.nuget.org/packages/Dax.ViewVpaExport/
https://www.nuget.org/packages/Dax.Model.Extractor/

The open source VertiPaq Analyzer project is here:
https://github.com/sql-bi/VertiPaq-Analyzer
